src/controllers/articleControllers.js

```diff
@@ -0,0 +1,62 @@
+import model from '../db/models/index';
+
+const { Users, Draftarticles, Rating } = model;
+
+/**
+ * rating controller
+ */
+class ArticleManager {
+  /**
+     *
+     * @param {Object} req
+     * @param {Object} res
+     * @returns {Object} return object
+     */
+  static async rateArticle(req, res) {
+    const { slug } = req.params;
+    const { rating } = req.body;
+
+    // Check if the article exists
+    const findArticle = await Draftarticles.findOne({ where: { slug } });
+    if (!findArticle) {
+      return res.status(404).send({
+        error: 'Article not found!'
+      });
+    }
+    // find a user in token if he/she exist in the database
+    const findUser = await Users.findOne({ where: { email: req.user.email } });
+
+    // check if article belongs to the login user
+    if (findUser.id === findArticle.author) {
+      return res.status(400).json({
+        error: 'You can not rate your article!'
+      });
+    }
+
+    const findIfUserIfRatedArticle = await Rating.findOne({
+      where: {
+        user: findUser.id,
+        post: findArticle.id
+      }
+    });
+    if (findIfUserIfRatedArticle) {
+      return res.status(400).json({
+        error: 'You are already rate this article'
```

Comment:
- can you change the message of this response?

Reply:
Yes I can. Review again was changed